kind/pkg/cluster/node.go

```diff
@@ -103,6 +106,28 @@ func (nh *nodeHandle) Run(command string, args ...string) error {
 	return cmd.Run()
 }
 
+// RunHook runs a LifecycleHook on the node
+// It will only return an error if hook.MustSucceed is true
+func (nh *nodeHandle) RunHook(hook *config.LifecycleHook, phase string) error {
+	logger := logrus.WithFields(logrus.Fields{
+		"node":  nh.nameOrID,
+		"phase": phase,
+	})
```

Comment:
yeah it is, i'm still very on the fence about this as well. it's useful for some things like error codes I think, but can be harder to read as a CLI user than carefully formatted messages I think, especially for information dense logs instead of "Foo failed because bar" you can wind up with "failure thing=foo because=bar error=blah node=abc" which is maybe not better? /shrug

Reply:
we still need a story about how we want to do really good logging. so far it's just _ok_